_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -6273,7 +6273,7 @@ mfxStatus ConfigureExecuteParams(
     }
 
 
-    if (true == executeParams.bSceneDetectionEnable && (FRC_ENABLED | config.m_extConfig.mode))
+    if (executeParams.bSceneDetectionEnable && (FRC_ENABLED & config.m_extConfig.mode))
```

Comment:
Hm. warning was that condition is always true, so did we execute unintended code sometimes? This change might require validation? which tests did you run?

Reply:
> did we execute unintended code sometimes?

Yes, I suppose. I think that condition like that: 
```cpp
    if (executeParams.bSceneDetectionEnable)
        executeParams.bSceneDetectionEnable = false;
```
doesn't make any sense. But I need to dive into investigation of these params.

Reply:
Actually, `bSceneDetectionEnable` is never used:
```
[dvrogozh@dvrscl msdk]$ fgrep -rsn bSceneDetectionEnable _studio/
_studio/mfx_lib/vpp/src/mfx_vpp_jpeg_d3d9.cpp:58:    pMfxExecuteParams->bSceneDetectionEnable = false;
_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp:131:    pMfxExecuteParams->bSceneDetectionEnable = false;
_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp:6181:                    executeParams.bSceneDetectionEnable = false;
_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp:6276:    if (true == executeParams.bSceneDetectionEnable && (FRC_ENABLED | config.m_extConfig.mode))
_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp:6279:        executeParams.bSceneDetectionEnable = false;
_studio/shared/include/mfx_vpp_interface.h:278:               ,bSceneDetectionEnable(false)
_studio/shared/include/mfx_vpp_interface.h:345:                    bSceneDetectionEnable != false ||
_studio/shared/include/mfx_vpp_interface.h:401:        bool           bSceneDetectionEnable;
```

So, the fix should be: delete them all! :)

Reply:
Done.